articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Login
+description: This tutorial demonstrates how to add user login, logout, and profile to a Node.js Express application.
+budicon: 448
+topics:
+  - quickstarts
+  - webapp
+  - login
+  - user profile
+  - logout
+  - nodejs
+contentType: tutorial
+useCase: quickstart
+github:
+  path: 01-Login
+---
+
+<!-- markdownlint-disable MD002 -->
+## Getting Started
+In this quickstart you will learn how to implement authentication in an Express application using the OIDC Compliant library [Express OIDC](https://github.com/auth0/express-openid-connect) authored and maintained by Auth0.
+
+::: warning
+Express OIDC is still in an [early access stage](/).
+:::
+
+### Install Dependencies
+
+```sh
+# Using NPM
+npm install express-openid-connect express-session
+
+# Using Yarn
+yarn add express-openid-connect express-session
+```
+
+## Configure Auth0
+### Configure Callback URLs
+
+A callback URL is a URL in your application where Auth0 redirects the user after they have authenticated. 
+
+The callback URL for your app must be whitelisted in the **Allowed Callback URLs** field in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings). If this field is not set, users will be unable to log in to the application and will get an error.
+
+::: note
+If you are testing your application locally, you should set the **Allowed Callback URL** to `http://localhost:3000/callback`
+:::
+
+### Configure Logout URLs
+
+A logout URL is a URL in your application that Auth0 can return to after the user has been logged out of the authorization server. This is specified in the `returnTo` query parameter.
+
+The logout URL for your app must be whitelisted in the **Allowed Logout URLs** field in your [Application Settings](${manage_url}/#/applications/${account.clientId}/settings). If this field is not set, users will be unable to log out from the application and will get an error.
+
+::: note
+If you are testing your application locally, the logout URL you need to whitelist in the **Allowed Logout URLs** field is `http://localhost:3000`.
+:::
+
+## Integrate Auth0
+### Auth0 Application Keys
+
+You will need to retrieve some details for your application to communicate with Auth0. You can get these details from the [Application Settings](${manage_url}/#/applications/${account.clientId}/settings) section in the Auth0 dashboard.
+
+You need the following information:
+
+* **Domain**
+* **Client ID**
+* **Client Secret**
+
+### Express Middleware
+There are three middlewares required to integrate Auth0 in your application.
+
+- `express.urlencoded` - parses urlencoded payloads sent back from the Auth0 server.
```

Comment:
Might want to mention that `bodyParser.urlencoded` works the same and might already be in use in an application. It's also the way to get this working with a lower version of Express.

Reply:
Do we have more documentation on that?  What version of express does it stop being required?

Reply:
The `urlencoded` always needs to be there but Express 4.16 added it to `express`. Before that version, you'd need to install `body-parser` and use that.

Reply:
The quickstart and the library call out that Express 4.16+ is required

Reply:
True but both work and they might already be using body parser. I don't know if it hurts to have both in there or not but it might be worth it just to say that both are acceptable. Or maybe thats TMI?